docs/intro/quick-start.md

```diff
@@ -0,0 +1,2826 @@
++    pub fn delete(ctx: Context<Delete>) -> Result<()> {
+         msg!("Delete Message");
+
++        let user_key = ctx.accounts.user.key();
++        let signer_seeds: &[&[&[u8]]] =
++            &[&[b"vault", user_key.as_ref(), &[ctx.bumps.vault_account]]];
++
++        let transfer_accounts = Transfer {
++            from: ctx.accounts.vault_account.to_account_info(),
++            to: ctx.accounts.user.to_account_info(),
++        };
++        let cpi_context = CpiContext::new(
++            ctx.accounts.system_program.to_account_info(),
++            transfer_accounts,
++        ).with_signer(signer_seeds);
++        transfer(cpi_context, ctx.accounts.vault_account.lamports())?;
+         Ok(())
+     }
+
+```
+
+</details>
+
+<details>
+{<summary>Explanation</summary>}
+
+In the delete instruction, we implement another Cross Program Invocation (CPI)
+to invoke the System Program's transfer instruction. This CPI demonstrates how
+to make a transfer that requires a Program Derived Address (PDA) signer.
+
+First, we define the signer seeds for the vault PDA:
+
+```rs filename="lib.ts"
+let user_key = ctx.accounts.user.key();
+let signer_seeds: &[&[&[u8]]] =
+    &[&[b"vault", user_key.as_ref(), &[ctx.bumps.vault_account]]];
+```
+
+The `Transfer` struct specifies the required accounts for the System Program's
+transfer instruction:
+
+- from: The vault account (source of funds)
+- to: The user's account (destination of funds)
+
+  ```rs filename="lib.ts"
+  let transfer_accounts = Transfer {
+      from: ctx.accounts.vault_account.to_account_info(),
+      to: ctx.accounts.user.to_account_info(),
+  };
+  ```
+
+The CpiContext specifies:
+
+- The program to be invoked (System Program)
+- The accounts involved in the transfer (defined in the Transfer struct)
+- The signer seeds for the PDA
+
+  ```rs filename="lib.ts"
+  let cpi_context = CpiContext::new(
+      ctx.accounts.system_program.to_account_info(),
+      transfer_accounts,
+  ).with_signer(signer_seeds);
+  ```
+
+The transfer function then invokes the transfer instruction on the System
+Program, passing:
+
+- The cpi_context (program, accounts, and PDA signer)
+- The amount to transfer (the entire balance of the vault account)
+
+  ```rs filename="lib.ts"
+  transfer(cpi_context, ctx.accounts.vault_account.lamports())?;
+  ```
+
+This CPI implementation demonstrates how programs can utilize PDAs to manage
+funds. When our program's delete instruction is invoked, it internally calls the
+System Program's transfer instruction, signing for the PDA to authorize the
+transfer of all funds from the vault back to the user.
+
+</details>
+
+Rebuild the program.
+
+```shell filename="Terminal" copy
+build
+```
+
+### 3. Redeploy Program
+
+After making these changes, we need to redeploy our updated program. This
+ensures that our modified program is available for testing. On Solana, updating
+a program simply requires deploying the compiled program at the same program ID.
```

Comment:
can you clarify that you also need to be the authority, which is normally your cli keypair

Reply:
added

```
<details>
{<summary>Explanation</summary>}

Only the upgrade authority of the program can update it. The upgrade authority
is set when the program is deployed, and it's the only account with permission
to modify or close the program. If the upgrade authority is revoked, then the
program becomes immutable and can never be closed or upgraded.

When deploying programs on Solana Playground, your Playground wallet is the
upgrade authority for all your programs.

</details>

```